python_modules/dagster/dagster_tests/asset_defs_tests/test_assets.py

```diff
@@ -743,17 +743,49 @@ def bar_manager():
             AssetSpec(key=AssetKey("key1"), metadata={SYSTEM_METADATA_KEY_IO_MANAGER_KEY: "foo"}),
             AssetSpec(key=AssetKey("key2"), metadata={SYSTEM_METADATA_KEY_IO_MANAGER_KEY: "bar"}),
         ],
-        resource_defs={"foo": foo_manager, "bar": bar_manager, "baz": baz_resource},
     )
     def my_asset(context):
-        # Required io manager keys are available on the context, same behavoir as ops
+        # Required io manager keys are available on the context, same behavior as ops
         assert hasattr(context.resources, "foo")
         assert hasattr(context.resources, "bar")
         yield Output(1, "key1")
         yield Output(2, "key2")
 
+    resource_dict = {"foo": foo_manager, "bar": bar_manager, "baz": baz_resource}
     with instance_for_test() as instance:
-        materialize([my_asset], instance=instance)
+        materialize([my_asset], instance=instance, resources=resource_dict)
+
+    assert foo_list == [1]
+    assert bar_list == [2]
+
+    foo_list = []
+    bar_list = []
+
+    # Introduce a downstream multi-asset which overrides the input manager for one asset, and uses the default for the other.
+    @io_manager
+    def override_manager():
+        class MyOverriddenIOManager(MyIOManager):
+            def load_input(self, _context):
+                return self._the_list[-1] + 1
+
+        return MyOverriddenIOManager(bar_list)
+
+    @multi_asset(
+        specs=[
```

Comment:
I agree that the ergonomics are pretty weird here. Has there been discussion that combining `specs` and `ins` is the right thing to start doing in these cases? From my understanding (based on discussions from months ago, so likely out of date) the idea was that `ins` and `outs` are used together, and `specs` was more intended for use cases that don't use the i/o manager. 

Although it does seem like a valid use case to have an asset that needs to load inputs with an i/o manager but does not store an output via an i/o manager, so combining ins and specs could be the right thing to do.

Overall i think this is a good test to add since we don't explicitly disallow this combo, so we should have it under test. I am curious if there are updates to how we are thinking AssetSpecs should be used though!